velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -436,6 +438,165 @@ TpchPlan TpchQueryBuilder::getQ6Plan() const {
+  const auto& partsuppFileColumns = getFileColumnNames(kPartsupp);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  auto nationSelectedRowType = getRowType(kNation, nationColumns);
+  const auto& nationFileColumns = getFileColumnNames(kNation);
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId partScanNodeId;
+  core::PlanNodeId supplierScanNodeId;
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId partsuppScanNodeId;
+  core::PlanNodeId ordersScanNodeId;
+  core::PlanNodeId nationScanNodeId;
+
+  auto part = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(
+                      kPart,
+                      partSelectedRowType,
+                      partFileColumns,
+                      {},
+                      "p_name like '%green%'")
+                  .capturePlanNodeId(partScanNodeId)
+                  .planNode();
+
+  auto supplier =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kSupplier, supplierSelectedRowType, supplierFileColumns)
+          .capturePlanNodeId(supplierScanNodeId)
+          .planNode();
+
+  auto orders =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kOrders, ordersSelectedRowType, ordersFileColumns)
+          .capturePlanNodeId(ordersScanNodeId)
+          .planNode();
+
+  auto nation =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kNation, nationSelectedRowType, nationFileColumns)
+          .capturePlanNodeId(nationScanNodeId)
+          .planNode();
+
+  auto lineitemJoinPart =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
+          .capturePlanNodeId(lineitemScanNodeId)
+          .hashJoin(
+              {"l_partkey"},
+              {"p_partkey"},
+              part,
+              "",
+              {"l_suppkey",
+               "l_partkey",
+               "l_orderkey",
+               "l_extendedprice",
+               "l_discount",
+               "l_quantity"})
+          .hashJoin(
+              {"l_suppkey"},
+              {"s_suppkey"},
+              supplier,
+              "",
+              {"l_suppkey",
+               "l_partkey",
+               "l_orderkey",
+               "l_extendedprice",
+               "l_discount",
+               "l_quantity",
+               "s_nationkey"})
+          .planNode();
+
+  auto partsuppJoinLineitem =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kPartsupp, partsuppSelectedRowType, partsuppFileColumns)
+          .capturePlanNodeId(partsuppScanNodeId)
+          .hashJoin(
+              {"ps_partkey", "ps_suppkey"},
+              {"l_partkey", "l_suppkey"},
+              lineitemJoinPart,
+              "",
+              {"l_orderkey",
+               "l_extendedprice",
+               "l_discount",
+               "l_quantity",
```

Comment:
Maybe add a local variable of a vector of the 4 column names as it is being repeated multiple times.